contrib/99designs/gqlgen/tracer_test.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package gqlgen
+
+import (
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+
+	"github.com/99designs/gqlgen/client"
+	"github.com/99designs/gqlgen/example/todo"
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/99designs/gqlgen/handler"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestImplementsTracer(t *testing.T) {
+	var _ graphql.Tracer = (*gqlTracer)(nil)
+}
+
+func newTodoClient(t graphql.Tracer) *client.Client {
+	c := client.New(handler.GraphQL(
+		todo.NewExecutableSchema(todo.New()),
+		handler.Tracer(t),
+	))
+	return c
+}
+
+// findRootSpan returns the first root span it finds in a slice of spans, or nil if none is found.
+func findRootSpan(spans []mocktracer.Span) mocktracer.Span {
+	for _, span := range spans {
+		if span.ParentID() == 0 {
+			return span
+		}
+	}
+	return nil
+}
+
+// findRootSpan returns the first span it finds with a matching tag value in a slice of spans, or nil if none is found.
+func findSpanWithTag(spans []mocktracer.Span, tag string, val interface{}) mocktracer.Span {
+	for _, span := range spans {
+		if span.Tag(tag) == val {
+			return span
+		}
+	}
+	return nil
+}
+
+func TestRootSpan(t *testing.T) {
+	for name, tt := range map[string]struct {
+		clientOpts []client.Option
+		tracerOpts []Option
+		test       func(assert *assert.Assertions, spans []mocktracer.Span)
```

Comment:
You only seem to be using the root span throughout and the call to `findRootSpan` is repeated everywhere, maybe it's better to:
```suggestion
		test       func(assert *assert.Assertions, root mocktracer.Span)
```
You can likely get rid of the function `findRootSpan` now and just inline it.